### PR TITLE
mocktikv: add rw mutex for mocktikv.Region

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/19443

Problem Summary:

data race ci error in `sendReqToRegion`, see detailed log in https://github.com/pingcap/tidb/issues/19443

The data race root cause analyzing is in the comment, see detail here:  https://github.com/pingcap/tidb/issues/19443#issuecomment-681908314

### What is changed and how it works?

Add rw mutex to control the concurrency visit for the mocktikv.Region

### Related changes

- Need to cherry-pick to the release branch  (release-4.0.0 / release-3.0.0)

### Check List 

Tests 

- Unit test

### Release note 

-

### CODE REVIEW

**Comment by Yisaer** [comment]
@SunRunAway  PTAL

**Comment by Yisaer** [comment]
@ichn-hu PTAL

**Review by ichn-hu — CHANGES_REQUESTED** [review]
@Yisaer Would you please  write a test for this?

**Comment by ichn-hu** [comment]
LGTM

**Comment by ichn-hu** [comment]
> @ichn-hu,Thanks for your review. However, LGTM is restricted to Reviewers or higher roles.You are not a reviewer or committer or co-leader or leader.

One more PR and I'll let you know who's the dad.

File: store/mockstore/mocktikv/cluster.go [hunk 1]
```diff
@@ -545,6 +545,7 @@ func (c *Cluster) getRegionsCoverRange(start, end MvccKey) []*Region {
 
 // Region is the Region meta data.
 type Region struct {
+	rw     sync.RWMutex
```
Comment (SunRunAway) [inline store/mockstore/mocktikv/cluster.go:548]: The `Cluster` which is the caller of `Region` already has a Lock, could it be unified with this Lock?

**Comment by Yisaer** [comment]
I forgot to paste the data race stack. From the stack here, it seems the mutex in the `Cluster`won't help to avoid data race.

```sh
WARNING: DATA RACE
Write at 0x00c0011dee08 by goroutine 80:
  github.com/pingcap/tidb/store/mockstore/mocktikv.(*Region).updateKeyRange()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/mockstore/mocktikv/cluster.go:569 +0x657
  github.com/pingcap/tidb/store/mockstore/mocktikv.(*Region).split()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/mockstore/mocktikv/cluster.go:559 +0x5be
  github.com/pingcap/tidb/store/mockstore/mocktikv.(*Cluster).SplitRaw()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/mockstore/mocktikv/cluster.go:327 +0x184
  github.com/pingcap/tidb/store/mockstore/mocktikv.(*rpcHandler).handleSplitRegion()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/mockstore/mocktikv/rpc.go:626 +0x48b
  github.com/pingcap/tidb/store/mockstore/mocktikv.(*RPCClient).SendRequest()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/mockstore/mocktikv/rpc.go:973 +0x50ed
[2020/09/02 15:04:08.243 +08:00] [INFO] [split_region.go:176] ["batch split regions, scatter region complete"] ["batch region ID"=20] [at="dIAAAAAAAAAzX3KAAAAAAAMNQA=="] ["new region left"="id:20 start_key:\"t\\200\\000\\000\\000\\000\\000\\000\\377/_r\\200\\000\\000\\000\\000\\377\\000#(\\000\\000\\000\\000\\000\\372\" region_epoch:<version:1 > peers:<id:21 store_id:1 > "]
  github.com/pingcap/tidb/store/tikv.(*RegionRequestSender).sendReqToRegion()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/tikv/region_request.go:177 +0x1d5
  github.com/pingcap/tidb/store/tikv.(*RegionRequestSender).SendReqCtx()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/tikv/region_request.go:131 +0x18e
  github.com/pingcap/tidb/store/tikv.(*RegionRequestSender).SendReq()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/tikv/region_request.go:74 +0x329
  github.com/pingcap/tidb/store/tikv.(*tikvStore).batchSendSingleRegion()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/tikv/split_region.go:125 +0x2bc
  github.com/pingcap/tidb/store/tikv.(*tikvStore).splitBatchRegionsReq.func1.1()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/tikv/split_region.go:79 +0x106
  github.com/pingcap/tidb/util.WithRecovery()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/util/misc.go:92 +0x68
  github.com/pingcap/tidb/store/tikv.(*tikvStore).splitBatchRegionsReq.func1()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/tikv/split_region.go:77 +0x196

Previous read at 0x00c0011dee08 by goroutine 31:
  github.com/golang/protobuf/proto.pointer.getPointer()
      /Users/yisa/Downloads/Github/GoProject/pkg/mod/github.com/golang/protobuf@v1.2.0/proto/pointer_unsafe.go:258 +0x8f5
  github.com/golang/protobuf/proto.(*mergeInfo).merge()
      /Users/yisa/Downloads/Github/GoProject/pkg/mod/github.com/golang/protobuf@v1.2.0/proto/table_merge.go:123 +0x18c
  github.com/golang/protobuf/proto.(*InternalMessageInfo).Merge()
      /Users/yisa/Downloads/Github/GoProject/pkg/mod/github.com/golang/protobuf@v1.2.0/proto/table_merge.go:50 +0x129
  github.com/pingcap/kvproto/pkg/metapb.(*Region).XXX_Merge()
      /Users/yisa/Downloads/Github/GoProject/pkg/mod/github.com/pingcap/kvproto@v0.0.0-20200311073257-e53d835099b0/pkg/metapb/metapb.pb.go:336 +0x64
  github.com/golang/protobuf/proto.Merge()
      /Users/yisa/Downloads/Github/GoProject/pkg/mod/github.com/golang/protobuf@v1.2.0/proto/clone.go:95 +0x28e
  github.com/golang/protobuf/proto.Clone()
      /Users/yisa/Downloads/Github/GoProject/pkg/mod/github.com/golang/protobuf@v1.2.0/proto/clone.go:52 +0x1b7
  github.com/pingcap/tidb/store/mockstore/mocktikv.(*rpcHandler).handleSplitRegion()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/mockstore/mocktikv/rpc.go:628 +0x4c4
[2020/09/02 15:04:08.246 +08:00] [INFO] [split_region.go:215] ["start scatter region"] [regionID=30]
  github.com/pingcap/tidb/store/mockstore/mocktikv.(*RPCClient).SendRequest()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/mockstore/mocktikv/rpc.go:973 +0x50ed
  github.com/pingcap/tidb/store/tikv.(*RegionRequestSender).sendReqToRegion()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/tikv/region_request.go:177 +0x1d5
  github.com/pingcap/tidb/store/tikv.(*RegionRequestSender).SendReqCtx()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/tikv/region_request.go:131 +0x18e
  github.com/pingcap/tidb/store/tikv.(*RegionRequestSender).SendReq()
      /Users/yisa/Downloads/Github/GoProject/src/github.com/pingcap/tidb/store/tikv/region_request.go:74 +0x329
  github.com/pingcap/tidb/store/tikv.(*tikvStore).batchSendSingleRegion()
      /Users/yisa/Downloads/G[2020/09/02 15:04:08.248 +08:00] [INFO] [split_region.go:176] ["batch split regions, scatter region complete"] ["batch region ID"=20] [at="dIAAAAAAAAAzX3KAAAAAAAYagA=="] ["new region left"="id:30 start_key:\"t\\200\\000\\000\\000\\000\\000\\000\\3773_r\\200\\000\\000\\000\\000\\377\\003\\r@\\000\\000\\000\\000\\000\\372\" region_epoch:<version:1 > peers:<id:31 store_id:1 > "]

```

**Pull request closed without merging**